File: android/revenuedog/src/main/kotlin/com/revenuedog/purchases/models/Offerings.kt
package com.revenuedog.purchases.models

/**
 * All offerings configured on the backend, resolved for this platform and subscriber
 * (experiment-aware: [current] already reflects any assigned experiment variant).
 */
class Offerings internal constructor(
    /** The offering marked current on the backend, or null when none is set. */
    val current: Offering?,
    /** All offerings keyed by identifier. */
    val all: Map<String, Offering>
) {
    /** Looks up an offering by identifier. */
    fun getOffering(identifier: String): Offering? = all[identifier]

    /** Operator sugar for [getOffering]. */
    operator fun get(identifier: String): Offering? = all[identifier]

    override fun toString(): String =
        "Offerings(current=${current?.identifier}, all=${all.keys})"
}

File: ios/Sources/RevenueDog/Models/Offerings.swift
import Foundation

/// The complete set of ``Offering``s configured for the app, plus the
/// currently-selected one.
public struct Offerings {
    /// The current offering, as resolved by the backend (experiment-aware), or
    /// `nil` if none is configured.
    public let current: Offering?
    /// All offerings keyed by identifier.
    public let all: [String: Offering]

    public init(current: Offering?, all: [String: Offering]) {
        self.current = current
        self.all = all
    }

    /// Look up an offering by identifier.
    public func offering(identifier: String) -> Offering? {
        all[identifier]
    }

    /// Look up an offering by identifier.
    public func get(_ identifier: String) -> Offering? {
        all[identifier]
    }

    /// Look up an offering by identifier via subscript.
    public subscript(_ identifier: String) -> Offering? {
        all[identifier]
    }
}
